infra/cifuzz/config_utils.py

```diff
@@ -88,7 +59,62 @@ def _get_language():
 # pylint: disable=too-few-public-methods,too-many-instance-attributes
 
 
-class ConfigurationError(Exception):
+class CiEnvGetter:
```

Comment:
Let's structure it like this:

```python

class CiEnvironment:
   WORKSPACE_VAR = 'WORKSPACE'
   REPOSITORY_VAR  = 'REPOSITORY'
   TOKEN_VAR = 'TOKEN'
   GIT_SHA_VAR = 'GIT_SHA'

  def workspace(self):
    return os.getenv(self.WORKSPACE_VAR)

  def owner_and_repo(self):
   repository = os.getenv(self.REPOSITORY_VAR)
   ...

  def project_src_path(self):
    return os.getenv('PROJECT_SRC_PATH')


class GitHubEnvironment(CiEnvironment):
   WORKSPACE_VAR = 'GITHUB_WORKSPACE'
   REPOSITORY_VAR  = 'GITHUB_REPOSITORY'
   TOKEN_VAR = 'GITHUB_TOKEN'
   GIT_SHA_VAR = 'GITHUB_SHA'

   def project_src_path(self):
      # On GitHub, they don't know the absolute path, it is relative to
      # |workspace|.
     return os.path.join(self.workspace(), super().project_src_path())
```

This way, there's a clearer separation and it becomes much easier to extend support for other CI systems in the future if necessary.

Reply:
Done. PTAL.